File: files/200-299/292/CH-292.R
library(tidyverse)
library(readxl)

path = "files/200-299/292/CH-292 Advanced Filtering.xlsx"
input = read_excel(path, range = "B2:E9")
test  = read_excel(path, range = "G2:G6")

result = input %>%
  rowwise() %>%
  filter(n_distinct(c_across(-ID)) == ncol(across(-ID))) %>%
  ungroup() %>%
  select(ID)
  
all.equal(result$ID, test$`Selected IDs`) # TRUE

File: files/200-299/292/CH-292.py
import pandas as pd

path = "200-299/292/CH-292 Advanced Filtering.xlsx"
input = pd.read_excel(path, usecols="B:E", skiprows=1, nrows=8)
test = pd.read_excel(path, usecols="G", skiprows=1, nrows=4)

result = input[input.drop('ID', axis=1).apply(lambda row: len(set(row)) == len(row), axis=1)][['ID']].reset_index(drop=True)

print(result['ID'].equals(test['Selected IDs'])) # True